orcid-pub-web/src/main/java/org/orcid/api/filters/ApiRateLimitFilter.java

```diff
@@ -0,0 +1,241 @@
+import org.orcid.persistence.jpa.entities.ProfileEntity;
+import org.orcid.persistence.jpa.entities.PublicApiDailyRateLimitEntity;
+import org.orcid.utils.email.MailGunManager;
+import org.orcid.utils.panoply.PanoplyPapiDailyRateExceededItem;
+import org.orcid.utils.panoply.PanoplyRedshiftClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import org.orcid.core.togglz.Features;
+
+@Component
+public class ApiRateLimitFilter extends OncePerRequestFilter {
+    private static Logger LOG = LoggerFactory.getLogger(ApiRateLimitFilter.class);
+
+    @Autowired
+    private PublicApiDailyRateLimitDao papiRateLimitingDao;
+
+    @Autowired
+    private OrcidSecurityManager orcidSecurityManager;
+
+    @Autowired
+    private ClientDetailsEntityCacheManager clientDetailsEntityCacheManager;
+
+    @Autowired
+    private MailGunManager mailGunManager;
+
+    @Autowired
+    private ProfileDao profileDao;
+
+    @Autowired
+    private OrcidUrlManager orcidUrlManager;
+
+    @Autowired
+    private RecordNameManager recordNameManager;
+
+    @Autowired
+    private TemplateManager templateManager;
+
+    @Autowired
+    private EmailManager emailManager;
+    
+    @Resource
+    private PanoplyRedshiftClient panoplyClient;  
+
+    @Value("${org.orcid.papi.rate.limit.anonymous.requests:10000}")
+    private int anonymousRequestLimit;
+
+    @Value("${org.orcid.papi.rate.limit.known.requests:40000}")
+    private int knownRequestLimit;
+
+    @Value("${org.orcid.papi.rate.limit.enabled:false}")
+    private boolean enableRateLimiting;
+    
+    @Value("${org.orcid.persistence.panoply.papiExceededRate.production:false}")
+    private boolean enablePanoplyPapiExceededRateInProduction; 
+
+    private static final String TOO_MANY_REQUESTS_MSG = "Too Many Requests - You have exceeded the daily allowance of API calls.\\n"
+            + "You can increase your daily quota by registering for and using Public API client credentials "
+            + "(https://info.orcid.org/documentation/integration-guide/registering-a-public-api-client/ )";
+
+    private static final String SUBJECT = "[ORCID] You have exceeded the daily Public API Usage Limit - ";
+    private static final String FROM_ADDRESS = "\"Engagement Team, ORCID\" <c.dumitru@orcid.org>";
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, FilterChain filterChain)
+            throws ServletException, IOException {
+        System.out.println("!!!!!!!!!!!!!!!!!!!!!!!   ApiRateLimitFilter starts, rate limit is : " + enableRateLimiting);
+        LOG.info("ApiRateLimitFilter starts, rate limit is : " + enableRateLimiting);
```

Comment:
@Camelia-Orcid could you please make this one a `trace` instead of `info`? 
Thanks